autosklearn/data/validation.py

```diff
@@ -0,0 +1,297 @@
+        )
+        return y
+
+    def _validate_dataframe_features(
+        self,
+        X: pd.DataFrame,
+    ) -> Union[pd.DataFrame, np.ndarray]:
+        """
+        Interprets a Pandas
+        Uses .iloc as a safe way to deal with pandas object
+        """
+
+        # Start with the features
+
+        # Register if a column needs encoding
+        enc_columns = []
+
+        # Also, register the feature types for the estimator
+        feature_types = []
+
+        # Make sure each column is a valid type
+        for i, column in enumerate(X.columns):
+            if X[column].dtype.name in self.valid_pd_enc_dtypes:
+                enc_columns.append(i)
+                feature_types.append('categorical')
+            elif not np.issubdtype(X[column].dtype, np.number):
+                if X[column].dtype.name == 'Object':
+                    raise ValueError(
+                        "Input Column {} has invalid type object. "
+                        "Cast it to a numerical value before using it in Auto-Sklearn."
+                        "You can use pandas.to_numeric() to do this".format(
+                            column,
+                        )
+                    )
+                elif pd.core.dtypes.common.is_datetime_or_timedelta_dtype(
+                    X[column].dtype
+                ):
+                    raise ValueError(
+                        "Input Column {} has invalid time-type. "
+                        "Please convert the time information to a categorical value. "
+                        "Usually, time information can be separated into a set of categorical "
+                        "features. For example, hour of the day can be converted to 24 "
+                        " categories.".format(
+                            column,
+                        )
+                    )
+                else:
+                    raise ValueError(
+                        "Input Column {} has unsupported dtype {}. "
+                        "Supported column types are categorical/bool/numerical dtypes. "
+                        "Make sure your data is formatted in a correct way, "
+                        "before feeding it to Auto-Sklearn.".format(
+                            column,
+                            X[column].dtype.name,
+                        )
+                    )
+            else:
+                feature_types.append('numerical')
+
+        # Make sure we only set this once. It should not change
+        if not self.feature_types:
+            self.feature_types = feature_types
+
+        # This proc has to handle multiple calls, for X_train
+        # and X_test scenarios. We have to make sure also that
+        # data is consistent within calls
+        if enc_columns:
+            if self.enc_columns and self.enc_columns != enc_columns:
+                raise ValueError(
+                    "Validation was already performed on input with enc columns={} "
```

Comment:
While this is all correct, a user might not know what's going on here as it talks about validation. However, the user invokes functions like `fit` and `predict`. Could you please rephrase this error message?

Reply:
rephrased like:
 raise ValueError(
                    "Changing the column-types of the input data to Auto-Sklearn is not "
                    "allowed. The estimator previously was fitted with categorical/boolean "
                    "columns {}, yet, the new input data has categorical/boolean values {}. "
                    "Please recreate the estimator from scratch when changing the input "
                    "data. ".format(
                        self.enc_columns
                        enc_columns,